src/sinks/aws_cloudwatch_logs/request.rs

```diff
@@ -66,7 +68,20 @@ impl Future for CloudwatchFuture {
         loop {
             match &mut self.state {
                 State::DescribeStream(fut) => {
-                    let response = try_ready!(fut.poll().map_err(CloudwatchError::Describe));
+                    let response = match fut.poll() {
+                        Ok(Async::Ready(res)) => res,
+                        Ok(Async::NotReady) => return Ok(Async::NotReady),
+                        Err(e) => {
+                            if let DescribeLogStreamsError::ResourceNotFound(_) = e {
```

Comment:
What happens if this means that the group exists but the stream doesn't?

Reply:
I checked this with the awscli and if a group does not exist it will return resource not found. If the stream doesn't exist it will return an empty list and a 200. This is the case for describelogstreams.

Reply:
Awesome